text_area.go

```diff
@@ -142,6 +169,41 @@ func (self *TextArea) atLineStart() bool {
 		(len(self.content) > self.cursor-1 && self.content[self.cursor-1] == '\n')
 }
 
+func (self *TextArea) atLineEnd() bool {
+	return self.atEnd() ||
+		(len(self.content) > self.cursor && self.content[self.cursor] == '\n')
+}
+
+func (self *TextArea) BackSpaceWord() {
+	if self.atLineStart() {
+		self.BackSpaceChar()
+		self.clipboard = "\n"
+		return
+	}
+
+	right := self.cursor
+	for !self.atLineStart() && strings.ContainsRune(WHITESPACES, self.content[self.cursor-1]) {
+		self.cursor--
+	}
+	separators := false
+	for !self.atLineStart() && strings.ContainsRune(WORD_SEPARATORS, self.content[self.cursor-1]) {
+		self.cursor--
+		separators = true
+	}
+	if !separators {
+		for !self.atLineStart() && !strings.ContainsRune(WHITESPACES+WORD_SEPARATORS, self.content[self.cursor-1]) {
+			self.cursor--
+		}
+	}
+
+	self.clipboard = string(self.content[self.cursor:right])
+	self.content = append(self.content[:self.cursor], self.content[right:]...)
+}
+
+func (self *TextArea) Yank() {
```

Comment:
oh now I see, yank types the string. I had assumed yank was more about copying the string

Reply:
@jesseduffield In emacs, 'paste' is called yank: e.g. [fzf](https://github.com/junegunn/fzf/blob/a06671b47f0284733c7edf7dae8f22f9758c8393/man/man1/fzf.1#L885)